.github/workflows/test_published_images.yml

```diff
@@ -0,0 +1,187 @@
+        timeout-minutes: 10
+        run: |
+          set -o errexit -o nounset -o pipefail ${RUNNER_DEBUG:+-x}
+
+          # shellcheck source=../.github/scripts/abstract-simple-smoke-test.sh
+          . .github/scripts/abstract-simple-smoke-test.sh
+
+          function simple-smoke-test() {
+            local organization=$1
+            local image_name=$2
+
+            local expected_jdk_version
+            if [ -n "${{ matrix.jdk-image-variant }}" ]; then
+              expected_jdk_version=${{ matrix.jdk-image-variant }}
+            else
+              expected_jdk_version=${{ inputs.EXPECTED_DEFAULT_JAVA_VERSION }}
+            fi
+
+            # Compute tag by concatenating tag_elements
+            .github/scripts/simple-smoke-test.sh "${organization}/${image_name}":$(IFS=- ; echo "${tag_elements[*]}") "${CONTAINER_NAME}" "${{ matrix.distribution_type }}" "${{ inputs.EXPECTED_HZ_VERSION }}" "${expected_jdk_version}"
+          }
+
+          case "${{ matrix.distribution_type }}" in
+            "oss")
+              image_name="hazelcast"
+              ;;
+            "ee")
+              image_name="hazelcast-enterprise"
+              ;;
+            *)
+              # Impossible as validated earlier
+              echoerr "Unrecognized distribution type ${{ matrix.distribution_type }}"
+              exit 1
+              ;;
+          esac
+
+          if [[ "${{ matrix.distribution_type }}" == "ee" ]]; then
+            export HZ_LICENSEKEY=${{ secrets.HZ_ENTERPRISE_LICENSE }}
+            export HZ_INSTANCETRACKING_FILENAME=instance-tracking.txt
+          fi
+
+          # To allow computing the required tag, store the elements in an array
+          tag_elements=("${{ inputs.IMAGE_VERSION }}")
+
+          if [[ -n "${{ matrix.variant }}" ]]; then
+            tag_elements+=("${{ matrix.variant }}")
+          fi
+
+          if [[ -n "${{ matrix.jdk-image-variant }}" ]]; then
+            tag_elements+=("jdk${{ matrix.jdk-image-variant }}")
+          fi
+
+          echo "Testing Docker registry"
+          simple-smoke-test hazelcast "${image_name}"
+
+          # Check additional EE repos
+          # Only populated for default variant, not "slim"
+          if [[ "${{ matrix.distribution_type }}" == "ee" && -z "${{ matrix.variant }}" ]]; then
+            # NLC repo only populated for absolute versions - not "latest", "latest-lts" etc tags
+            # Identify absolute version based on earlier parsing of version number
+            if [[ -n "${{ steps.version.outputs.major }}" ]]; then
+              echo "Testing NLC"
+              simple-smoke-test "${{ secrets.NLC_REPOSITORY }}/hazelcast_cloud" "hazelcast-nlc"
+            fi
+
+            echo "Testing Red Hat Catalog"
+            simple-smoke-test "registry.connect.redhat.com/hazelcast" "${image_name}-${{ steps.version.outputs.major }}-rhel8"
+          fi
+
+      - name: Get docker logs
+        if: ${{ failure() }}
```

Comment:
Is container still running at this point? 

Also, personally, I would always get the logs, just to have an option to double-check if everything is fine

Reply:
> Is container still running at this point?

It [_should_ have stopped](https://github.com/hazelcast/hazelcast-docker/blob/a898ae7781069744605d6d895c197b76ca31dbb2/.github/scripts/simple-smoke-test.sh#L72), but in [other places](https://github.com/hazelcast/hazelcast-docker/blob/a898ae7781069744605d6d895c197b76ca31dbb2/.github/workflows/build-pr.yml#L113) we don't check

> Also, personally, I would always get the logs, just to have an option to double-check if everything is fine

I don't have a strong opinion either way.

https://github.com/hazelcast/hazelcast-docker/pull/814/commits/ffbc09f18f16ce485e16f350be73d99a0049de87

Reply:
> > Is container still running at this point?
> 
> It [_should_ have stopped](https://github.com/hazelcast/hazelcast-docker/blob/a898ae7781069744605d6d895c197b76ca31dbb2/.github/scripts/simple-smoke-test.sh#L72), but in [other places](https://github.com/hazelcast/hazelcast-docker/blob/a898ae7781069744605d6d895c197b76ca31dbb2/.github/workflows/build-pr.yml#L113) we don't check

sorry, I thought logs are not accessible after docker stop but they are